R/render.R

```diff
@@ -882,6 +882,10 @@ render <- function(input,
 
       # if Lua filters are provided, add the command line switch
       if (!is.null(lua_filters <- output_format$pandoc$lua_filters)) {
+        if (is.na(Sys.getenv("RMARKDOWN_LUA_SHARED", unset = NA))) {
+          lua_env_vars <- xfun::set_envvar(c(RMARKDOWN_LUA_SHARED = pkg_file_lua("shared.lua")))
+          on.exit(xfun::set_envvar(lua_env_vars), add = TRUE)
+        }
```

Comment:
Maybe it is not anymore needed to check that this variable has been set. 
Can we decide that it is reserved for rmarkdown used ? 

The only benefit to keep that this way would be that we could set this environment variable to another value inside another package and `render()` would not change it but this would make our filter fail ?

I am still hesitant to do that in `render()` because Lua filters are mainly tied to an output format directly and this env var is required only our filter. There is no harm to make it available to every filter though - it is more lua helpers available. 

So as this is mandatory for our filter to work, it would not be good if this variable would be set to another value.

```suggestion
        lua_env_vars <- xfun::set_envvar(c(RMARKDOWN_LUA_SHARED = pkg_file_lua("shared.lua")))
        on.exit(xfun::set_envvar(lua_env_vars), add = TRUE)
```

This is why I am taking time on this...

Reply:
We could also imagine something more generic like
```suggestion
        shared <- attr(lua_filters, "shared")
        if (!is.null(shared)) {
          lua_env_vars <- xfun::set_envvar(c(RMARKDOWN_LUA_SHARED = shared))
          on.exit(xfun::set_envvar(lua_env_vars), add = TRUE)
        }
```
We could add the shared lua filter as an attributes when passing the Lua filters in `lua_filters=` argument in `output_formats`. 
But this would mean make sure to keep the attributes in `merge_pandoc_options()` - what to do if both base and overlay formats defines lua filters with an attributes in that case. 🤔

Reply:
> Maybe it is not anymore needed to check that this variable has been set.
> Can we decide that it is reserved for rmarkdown used ?

Yes we can.